frontend/src/app/forms/forms.page.ts

```diff
@@ -279,4 +319,63 @@ export class FormsPage implements OnInit, OnDestroy {
     }
     this.progressPercentage = validSteps / totalNumberOfRequiredSteps;
   }
+
+  /**
+   * Showing an alert message, when leaving a form page.
+   *
+   * returning a Promise<boolean>
+   * returning true, if user wants to leave the page.
+   * returning false, if user cancels leaving the page.
```

Comment:
My suggestion: Replace this with

```
@returns A Promise. This Promise will emit true if the user wants to leave the page. Otherwise, false is emitted.
```

Then you don't need this "returning, returning, returning".